geoh5py/objects/text.py

```diff
@@ -0,0 +1,225 @@
+    starting_point: str = Field(
+        validation_alias=AliasChoices("starting_point", "StartingPoint"),
+        serialization_alias="StartingPoint",
+    )
+    annotation: int = Field(
+        validation_alias=AliasChoices("annotation", "Annotation"),
+        serialization_alias="Annotation",
+        default=0,
+    )
+    color: str | None = Field(
+        validation_alias=AliasChoices("color", "Color"),
+        serialization_alias="Color",
+        default="#ffffff",
+    )
+    direction: str | None = Field(
+        validation_alias=AliasChoices("direction", "Direction"),
+        serialization_alias="Direction",
+        default="{1,0,0}",
+    )
+    font: str | None = Field(
+        validation_alias=AliasChoices("font", "Font"),
+        serialization_alias="Font",
+        default="builtin_simple",
+    )
+    font_size: int | None = Field(
+        validation_alias=AliasChoices("font_size", "FontSize"),
+        serialization_alias="FontSize",
+        default=10,
+    )
+    layer_name: str | None = Field(
+        validation_alias=AliasChoices("layer_name", "LayerName"),
+        serialization_alias="LayerName",
+        default="",
+    )
+    normal: str | None = Field(
+        validation_alias=AliasChoices("normal", "Normal"),
+        serialization_alias="Normal",
+        default="{0,0,1}",
+    )
+    obliquing_angle: int | None = Field(
+        validation_alias=AliasChoices("obliquing_angle", "ObliquingAngle"),
+        serialization_alias="ObliquingAngle",
+        default=0,
+    )
+    rotation_angle: int | None = Field(
+        validation_alias=AliasChoices("rotation_angle", "RotationAngle"),
+        serialization_alias="RotationAngle",
+        default=0,
+    )
+    text: str | None = Field(
+        validation_alias=AliasChoices("text", "Text"),
+        serialization_alias="Text",
+        default="<no-data>",
+    )
+    width_scale_factor: float | None = Field(
+        validation_alias=AliasChoices("width_scale_factor", "WidthScaleFactor"),
+        serialization_alias="WidthScaleFactor",
+        default=1.0,
+    )
+
+    @field_validator("starting_point", "normal", "direction", mode="before")
+    @classmethod
+    def array_to_str(cls, float_list: np.ndarray | list):
+        if isinstance(float_list, np.ndarray | list):
+            float_list = "{" + ",".join([str(val) for val in float_list]) + "}"
+
+        return float_list
+
+
+class TextData(BaseModel):
+    """
+    Text mesh data.
+    """
+
+    text_data: list[TextEntry] = Field(
+        validation_alias=AliasChoices("text_data", "Text Data"),
+        serialization_alias="Text Data",
+    )
+
+
+class TextObject(Points):
+    """
+    Text object for annotated text labels in viewport.
+    """
+
+    _attribute_map: dict = Points._attribute_map.copy()
+    _attribute_map.update({"TextMesh Data": "text_mesh_data"})
+    _default_name = "Text Object"
+    _TYPE_UID: uuid.UUID | None = uuid.UUID("{c00905d1-bc3b-4d12-9f93-07fcf1450270}")
```

Comment:
why is there an `| None` in the annotation?

Reply:
Bad copy I guess. Removing